src/couch/src/couch_file.erl

```diff
@@ -272,14 +286,18 @@ deleted_filename(Original) ->
             ++ filename:extension(Original), [Y, Mon, D, H, Min, S])),
     filename:rootname(Original) ++ Suffix.
 
-nuke_dir(RootDelDir, Dir) ->
+nuke_dir(RootDir, Dir) ->
+    nuke_dir(RootDir, Dir, []).
+nuke_dir(RootDelDir, Dir, Options) ->
     EnableRecovery = config:get_boolean("couchdb",
         "enable_database_recovery", false),
+    Context = couch_util:get_value(context, Options, compaction),
     case EnableRecovery of
-        true ->
-            rename_file(Dir);
-        false ->
-            delete_dir(RootDelDir, Dir)
+        true when Context == delete ->
+            DbName = couch_util:get_value(db_name, Options),
+            rename_dir(RootDelDir, Dir, DbName);
+        true -> rename_file(Dir);
```

Comment:
```
   case EnableRecovery of
        true when Context == delete ->
            DbName = couch_util:get_value(db_name, Options),
            rename_dir(RootDelDir, Dir, DbName);
        true -> rename_file(Dir);
        false -> delete_dir(RootDelDir, Dir)
```
From above, you can see that there are three conditions:
 - the first one: `EnableRecovery == true` and `Context == delete`
    under this situation, we want to move index to .recovery directory
 - the third one: `EnableRecovery == false`
    under this situation, we want to delete the file (either delete in place or move it to .delete directory

If we are talking about the second situation where `EnableRecovery == true` and `Context == compact`,  we still want to keep original logic, i.e. rename view file in place.